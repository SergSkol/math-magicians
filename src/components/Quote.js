import React from 'react';
import PropTypes from 'prop-types';

const arrQuotes = [
  'Math is fun. ...',
  'Pure mathematics is the world`s best game. ...',
  'Mathematics consists of proving the most obvious thing in the least obvious way. ...',
  'In mathematics, you don`t understand things. ...',
  'There are two ways to do great mathematics. ...',
  'Five out of four people have trouble with fractions. ...',
  'Mathematics is a hard thing to love.',
];

const getRandomIndex = () => {
  const random = Math.random();
  const randomIndex = Math.floor(arrQuotes.length * random);
  return randomIndex;
};

const Quote = (props) => {
  const { quoteIndex } = props;

  const GenerateQuote = () => {
    const quote = [];

    const index = quoteIndex || getRandomIndex();

    quote.push(<p key={index}> {arrQuotes[index]} </p>);
    return (quote);
  };

  return (
    <div className="quoteContent">
      <GenerateQuote />
    </div>
  );
};

Quote.defaultProps = {
  quoteIndex: getRandomIndex(),
};

Quote.propTypes = {
  quoteIndex: PropTypes.number,
};

export default Quote;
