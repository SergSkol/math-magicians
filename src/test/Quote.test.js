import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

it('Quote component snapshots test', () => {
  const tree = renderer
    .create(<Quote quoteIndex={5}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
