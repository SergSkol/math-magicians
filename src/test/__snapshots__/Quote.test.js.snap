// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component snapshots test 1`] = `
<div
  className="quoteContent"
>
  <p>
    <p>
       
      There are two ways to do great mathematics. ...
       
    </p>
  </p>
</div>
`;
